pkg/eval/str/str.go

```diff
@@ -92,6 +95,74 @@ import (
 // ▶ $false
 // ```
 
+//elvdoc:fn from-codepoints
+//
+// ```elvish
+// str:from-codepoints $number...
+// ```
+//
+// Outputs a string consisting of the given Unicode codepoints. Example:
+//
+// ```elvish-transcript
+// ~> str:from-codepoints 0x61
+// ▶ a
+// ~> str:from-codepoints 0x4f60 0x597d
+// ▶ 你好
+// ```
+//
+// @cf str:to-codepoints
+
+func fromCodepoints(nums ...int) (string, error) {
+	var b bytes.Buffer
+	for _, num := range nums {
+		if int(rune(num)) != num {
+			return "", errs.OutOfRange{
+				What:     "codepoint",
+				ValidLow: -2147483648, ValidHigh: 2147483647,
+				Actual: strconv.Itoa(num)}
+		}
+		if !utf8.ValidRune(rune(num)) {
+			return "", fmt.Errorf("invalid codepoint: %d", num)
```

Comment:
Return a BadValue struct here (https://github.com/elves/elvish/blob/a3f66b52af3c659af21596f0d44b76b2f41a0048/pkg/eval/errs/errs.go#L27)

Yoy can use "argument to str:from-codepoints" as "What" and "valid Unicode codepoint" as "Valid".